app/models/manageiq/providers/embedded_terraform/automation_manager/google_credential.rb

```diff
@@ -0,0 +1,71 @@
+class ManageIQ::Providers::EmbeddedTerraform::AutomationManager::GoogleCredential < ManageIQ::Providers::EmbeddedTerraform::AutomationManager::CloudCredential
+  COMMON_ATTRIBUTES = [
+    {
+      :component  => 'text-field',
+      :label      => N_('Service Account Email Address'),
+      :helperText => N_('The email address assigned to the Google Compute Engine service account'),
+      :name       => 'userid',
+      :id         => 'userid',
+      :type       => 'email',
+      :isRequired => true,
+      :validate   => [{:type => 'required'}],
+    },
+  ].freeze
+
+  EXTRA_ATTRIBUTES = [
+    {
+      :component      => 'password-field',
+      :label          => N_('RSA Private Key'),
+      :helperText     => N_('Contents of the PEM file associated with the service account email'),
+      :componentClass => 'textarea',
+      :name           => 'ssh_key_data',
+      :id             => 'ssh_key_data',
+      :type           => 'password',
+      :isRequired     => true,
+      :validate       => [{:type => 'required'}],
+    },
+    {
+      :component  => 'text-field',
+      :label      => N_('Project'),
+      :helperText => N_('The GCE assigned identification. It is constructed as two words followed by a three digit number, such as: squeamish-ossifrage-123'),
+      :name       => 'project',
+      :id         => 'project',
+      :maxLength  => 100,
+    },
+    {
+      :component  => 'text-field',
+      :label      => N_('Region'),
+      :helperText => N_('The default region for the resources. If another region is specified on the resource, it will take precedence.'),
+      :name       => 'region',
+      :id         => 'region',
+      :maxLength  => 50,
+    },
+  ].freeze
+
+  API_ATTRIBUTES = (COMMON_ATTRIBUTES + EXTRA_ATTRIBUTES).freeze
+
+  API_OPTIONS = {
+    :type       => 'cloud',
+    :label      => N_('Google Compute Engine'),
+    :attributes => API_ATTRIBUTES
+  }.freeze
+
+  alias ssh_key_data auth_key
+
+  def self.display_name(number = 1)
+    n_('Credential (Google)', 'Credentials (Google)', number)
+  end
+
+  def self.params_to_attributes(params)
+    attrs = super.dup
+
+    attrs[:auth_key] = attrs.delete(:ssh_key_data) if attrs.key?(:ssh_key_data)
```

Comment:
Basically the same question here as above (and looks like for a number of these).  Maybe there is a good reason? IDK

Reply:
Again same pattern as in ansible implementation ...

https://github.com/ManageIQ/manageiq/blob/master/app/models/manageiq/providers/embedded_ansible/automation_manager/google_credential.rb#L54

Reply:
Right, I don't think we should blindly copy&paste from other code without thinking about it first :)

Reply:
I'll get this for followup.